enhancements/update/eus-upgrades-mvp.md

```diff
@@ -0,0 +1,366 @@
+upgrade path from 4.6 to 4.10 so that they may mirror all content necessary
+- We will provide guidance on expected upgrade duration scaled by node count exclusive
+of the variability associated with workload rescheduling (ie: everything except
+Worker MCP rollout)
+- We will inhibit upgrades to the next minor version whenever necessary to preserve
+cluster stability and supportability, ie:
+  - Respecting version skew policies between OS managed components (kubelet, crio, RHCOS)
+  and operator managed components (Kubernetes API, OpenShift API, etc)
+  - Respecting version skew policies of OLM managed Operators
+- We will ensure that CVO managed operators upgrade as quickly as is safe to do so
+- We will reduce workload disruption associated with pod rescheduling during rolling
+reboots
+
+
+### Non-Goals
+
+- This enhancement does not attempt to remove *any* steps along the serial
+  upgrade path from 4.6 to 4.7 to 4.8 to 4.9 to 4.10 *including* reboots. A
+  follow-up enhancement will address those concerns.
+
+## Proposal
+
+Before we dive into user stories it's useful that you're familiar with CVO and
+Console feature which are relevant to how upgrades and channel options are
+presented to the admin. The CVO and Console as of 4.6 are now aware of other
+channels containing the cluster's current version. This means that if 4.6.50 is
+a non EUS version and 4.6.51 is an EUS version both the CVO and Console can
+display a different set of channel options. This means that 4.6.50 would see
+candidate-4.6, fast-4.6, stable-4.6, eus-4.6, candidate-4.7, fast-4.7, and
+stable-4.7 channels.
+
+Where as 4.6.51 would see only eus-4.6 and eus-4.10 assuming we have enabled EUS
+4.6 to 4.10 upgrades. After switching to eus-4.10 the cluster would offer
+upgrades to 4.7 versions present in the eus-4.10 channel, when that completes
+the cluster would offer 4.8 versions present in the eus-4.10 channel, repeat
+until the cluster has completed its upgrade to 4.10.
+
+### User Stories
+
+#### As an admin preparing for 4.6 to 4.10 EUS upgrade I'm alerted to API removals
+
+We have pre-existing mechanisms to alert admins to their use of APIs which are
+slated for removal in a future release. Currently those mechanisms do not look
+forward far enough to span 4.6 to 4.10, we will need to add the ability to
+backport that API metadata to 4.6 EUS versions once we have a firm understanding
+of all removals between 4.6 and 4.10.
+
+#### As an admin I cannot violate OpenShift's defined Kubelet to API version skew
+
+The MCO, or another suitable component, must set Upgradeable=False in order to
+prevent upgrades whenever a managed node would violate Kubelet to API version
+skew defined by OpenShift. For instance, assume OCP 4.6, 4.7, and 4.8 map to
+Kubernetes versions 1.19, 1.20, 1.21. If OpenShift chooses to support only
+version skew of N-1 the MCO would inhibit upgrades to 4.8 until all managed
+nodes had upgraded to 4.7.
```

Comment:
I *think* what is being requested here is that we finish the "Wait for all pools" work where we block upgrades until everything has been upgraded; we're planning on that work finishing in 4.8.

I am leary about the MCO gaining any awareness of versions or brokering if a specific version is compatible with the running Kubelet versions. A better path would for all components to report their API versions and the CVO blocks upgrades until the components show the compatible state.

Reply:
I think the MCO is the right Operator but we can be flexible on implementation for how MCO knows when to inhibit minor version upgrades. No other operator manages the OS and thus manages the Kubelet.

Reply:
I've updated this proposed epic to reflect what we discussed regarding constraints being defined by policy and enforced by MCO.